src/main/java/org/csanchez/jenkins/plugins/kubernetes/PodTemplate.java

```diff
@@ -222,50 +260,97 @@ public void setImagePullSecrets(List<PodImagePullSecret> imagePullSecrets) {
         this.imagePullSecrets.addAll(imagePullSecrets);
     }
 
+    @Deprecated
     public String getResourceRequestMemory() {
-        return resourceRequestMemory;
+        return getFirstContainer().map(ContainerTemplate::getResourceRequestMemory).orElse(null);
     }
 
+    @Deprecated
     @DataBoundSetter
     public void setResourceRequestMemory(String resourceRequestMemory) {
-        this.resourceRequestMemory = resourceRequestMemory;
+        getFirstContainer().ifPresent((i) -> i.setResourceRequestMemory(resourceRequestMemory));
     }
 
+    @Deprecated
     public String getResourceLimitCpu() {
-        return resourceLimitCpu;
+        return getFirstContainer().map(ContainerTemplate::getResourceLimitCpu).orElse(null);
     }
 
+    @Deprecated
     @DataBoundSetter
     public void setResourceLimitCpu(String resourceLimitCpu) {
-        this.resourceLimitCpu = resourceLimitCpu;
+        getFirstContainer().ifPresent((i) -> i.setResourceLimitCpu(resourceLimitCpu));
     }
 
+    @Deprecated
     public String getResourceLimitMemory() {
-        return resourceLimitMemory;
+        return getFirstContainer().map(ContainerTemplate::getResourceLimitMemory).orElse(null);
     }
 
+    @Deprecated
     @DataBoundSetter
     public void setResourceLimitMemory(String resourceLimitMemory) {
-        this.resourceLimitMemory = resourceLimitMemory;
+        getFirstContainer().ifPresent((i) -> i.setResourceLimitMemory(resourceLimitMemory));
     }
 
+    @Deprecated
     public String getResourceRequestCpu() {
-        return resourceRequestCpu;
+        return getFirstContainer().map(ContainerTemplate::getResourceRequestCpu).orElse(null);
     }
 
+    @Deprecated
     @DataBoundSetter
     public void setResourceRequestCpu(String resourceRequestCpu) {
-        this.resourceRequestCpu = resourceRequestCpu;
+        getFirstContainer().ifPresent((i) -> i.setResourceRequestCpu(resourceRequestCpu));
     }
 
+    @DataBoundSetter
+    public void setVolumes(@Nonnull List<PodVolume> items) {
+        synchronized (this.volumes) {
+            this.volumes.clear();
+            this.volumes.addAll(items);
+        }
+    }
+
+    @Nonnull
     public List<PodVolume> getVolumes() {
         return volumes;
     }
 
+    @DataBoundSetter
+    public void setContainers(@Nonnull List<ContainerTemplate> items) {
+        synchronized (this.containers) {
+            this.containers.clear();
+            this.containers.addAll(items);
+        }
+    }
+
+    @Nonnull
     public List<ContainerTemplate> getContainers() {
         return containers;
     }
 
+    @SuppressWarnings("deprecation")
```

Comment:
So, if I get it right, this is the proper way to create the fallback container, right?